iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/ArmTreeUtils.java

```diff
@@ -112,4 +116,10 @@ public static Predicate<Expression> containsParameterReference(Collection<String
       hasIdentifier.identifier() instanceof Identifier identifier &&
       parameterNames.contains(identifier.value());
   }
+
+  public static Optional<Property> getResourceProperty(ResourceDeclaration resourceDeclaration, String key) {
```

Comment:
Maybe it can be moved to `ResourceDeclaration`? This will be a more Object Oriented design